### PR TITLE
docs: Revamp storage doc

### PR DESCRIPTION
**Pull request by fenos**
## What kind of change does this PR introduce?

docs update

## What is the new behavior?

Revamp Storage Docs

### CODE REVIEW

File: apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+### Upload URL
+
+When uploading using the resumable upload endpoint, the TUS server creates a unique URL for each upload, even for multiple uploads to the same path. All chunks will be uploaded to this URL using the `PATCH` method.
+
+This URL will be valid for **up to 24 hours**. If the upload is not completed within 24 hours, the URL will expire and you'll need to start the upload again. The TUS client library will automatically create a new URL if the previous one expires.
+
+### Concurrency
+
+When two or more clients try to upload to the same Upload URL only one of them will succeed. The other clients will receive a `409 Conflict` error. Only 1 client can upload to the same Upload URL at a time which prevents data corruption.
```
Comment (gregnr) [inline apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx:129]:
```suggestion
When two or more clients try to upload to the same upload URL only one of them will succeed. The other clients will receive a `409 Conflict` error. Only 1 client can upload to the same upload URL at a time which prevents data corruption.
```

File: apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+When two or more clients upload a file to the same path using different upload URLs, the first client to complete the upload will succeed and the other clients will receive a `409 Conflict` error.
+
+If you provide the `x-upsert` header the last client to complete the upload will succeed instead.
+
+### UppyJS Example
+
+You can check a full example using UppyJS [Here](https://github.com/supabase/supabase/tree/master/examples/storage/resumable-upload-uppy)
```
Comment (gregnr) [inline apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx:137]:
```suggestion
You can check a [full example using UppyJS](https://github.com/supabase/supabase/tree/master/examples/storage/resumable-upload-uppy).
```

File: apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+Framework integration for UppyJS:
```
Comment (gregnr) [inline apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx:139]:
```suggestion
Below are the different framework integrations for UppyJS:
```

File: apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+- [Svelte](https://uppy.io/docs/svelte/)
+- [Vue](https://uppy.io/docs/vue/)
+- [Angular](https://uppy.io/docs/angular/)
+
+## Overwriting Files
+
+When uploading a file to a path that already exists, the default behavior is to return a `400 Asset Already Exists` error.
+If you want to overwrite a file on a specific path you can set the `x-upsert` header to `true`.
+
+We do advise against overwriting files when possible, as the CDN will take sometime to propagate the changes to all the edge nodes leading to stale content.
```
Comment (gregnr) [inline apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx:151]:
```suggestion
We do advise against overwriting files when possible, as the CDN will take some time to propagate the changes to all the edge nodes leading to stale content.
```

File: apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+Uploading a file to a new path is the recommended way to avoid propagation delays and stale content.
+
+If you want to know more about our CDN, check the [CDN](/docs/guides/storage/cdn) guide.
```
Comment (gregnr) [inline apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx:154]:
```suggestion
To learn more, see the [CDN](/docs/guides/storage/cdn) guide.
```

File: apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+## Access Control with RLS Policies
+
+Both the standard file upload and TUS resumable file upload methods in Supabase Storage support Row-Level Security (RLS) policies. RLS allows you to define fine-grained access control rules to restrict access to files based on user roles, permissions, or other criteria. This ensures that your files are secure and accessible only to authorized users.
+
+  For more information on RLS policies, see the [Storage Access Control](/docs/guides/storage/access-control) guide.
```
Comment (gregnr) [inline apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx:160]:
```suggestion
For more information on RLS policies, see the [Storage Access Control](/docs/guides/storage/access-control) guide.
```

File: apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+## Conclusion
+
+Supabase Storage provides multiple options for uploading files, including standard file upload and TUS resumable file upload, both of which use RLS policies for access control.
+  Additionally, there are client-side libraries like Uppy-js that make it easy to implement TUS uploads in your application. Choose the upload method that best fits your needs based on file size, network stability, and other requirements.
```
Comment (gregnr) [inline apps/docs/pages/guides/storage/uploads/resumable-uploads.mdx:165]:
```suggestion
Additionally, there are client-side libraries like Uppy-js that make it easy to implement TUS uploads in your application. Choose the upload method that best fits your needs based on file size, network stability, and other requirements.
```

File: apps/docs/pages/guides/storage/uploads/standard-uploads.mdx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+The standard file upload method is ideal for small files that are not larger than 6MB.
+
+It uses the traditional `multipart/form-data` format and is simple to implement using the supabase-js SDK. Here's an example of how to upload a file using the standard upload method:
+
+<Admonition type="note">
+  Though you can upload up to 5GB files using the standard upload method, we recommend use [TUS
+  Resumable Upload](/guides/storage/uploads/resumable-upload) for uploading files greater than 6MB in size for better
+  reliability.
+</Admonition>
```
Comment (gregnr) [inline apps/docs/pages/guides/storage/uploads/standard-uploads.mdx:21]:
```suggestion
<Admonition type="note">

Though you can upload up to 5GB files using the standard upload method, we recommend use [TUS Resumable Upload](/guides/storage/uploads/resumable-upload) for uploading files greater than 6MB in size for better reliability.
  
</Admonition>
```

**Review by gregnr — COMMENTED** [review]
Nice work @fenos!

**Review by MildTomato — APPROVED** [review]
approving only the redirect changes

File: apps/docs/pages/guides/storage/buckets/creating-buckets.mdx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+const { data, error } = await supabase.storage.createBucket('avatars', {
+  public: true,
+  allowedMimeTypes: ['image/*'],
+  maxFileSize: '1MB',
+})
+```
+
+If an upload request doesn't meet the above restrictions it will be rejected.
+
+For more information check the [File Validation](/docs/guides/storage/uploads/file-validation) Section.
```
Comment (fenos) [inline apps/docs/pages/guides/storage/buckets/creating-buckets.mdx:89]:
```suggestion
For more information check [File Limits](/docs/guides/storage/uploads/file-limits) Section.
```

File: apps/docs/pages/guides/storage/security/access-control.mdx [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+  tocVideo: '4ERX__Y908k',
+}
+
+Supabase Storage is designed to work perfectly with Postgres [Row Level Security](https://supabase.com/docs/guides/database/postgres/row-level-security) (RLS).
+
+You can use RLS to create [Security Access Policies](https://www.postgresql.org/docs/current/sql-createpolicy.html) that are incredibly powerful and flexible, allowing you to restrict access based on your business needs.
+
+## Access Policies
+
+By default Storage does not allow any uploads to private buckets without RLS policies. You selectively, allow certain operations by creating RLS policies on the `storage.objects` table.
```
Comment (fenos) [inline apps/docs/pages/guides/storage/security/access-control.mdx:17]:
```suggestion
By default Storage does not allow any uploads to buckets without RLS policies. You selectively, allow certain operations by creating RLS policies on the `storage.objects` table.
```

File: apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts [hunk 11]
```diff
@@ -975,18 +975,65 @@ export const storage: NavMenuConstant = {
...
+        { name: 'Error Codes', url: '/guides/storage/debugging/error-codes' },
+      ],
+    },
+    {
+      name: 'Schema',
+      url: undefined,
+      items: [
+        { name: 'Database Design', url: '/guides/storage/schema/design' },
+        { name: 'Helper Functions', url: '/guides/storage/schema/helper-functions' },
+        { name: 'Scaling', url: '/guides/storage/schema/scaling' },
```
Comment (fenos) [inline apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:1030]:
```suggestion

```

**Review by gregnr — APPROVED** [review]

**Pull request merged**